src/modulation/lora.rs

```diff
@@ -0,0 +1,53 @@
+//! Common LoRa modulation options
+
+/// LoRa mode channel configuration
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+
+pub struct LoRaChannel {
+    /// LoRa frequency in kHz
+    pub freq_khz: u32,
+    /// LoRa channel bandwidth
+    pub bw_khz: u16,
+    /// LoRa Spreading Factor
+    pub sf: SpreadingFactor,
+    /// LoRa Coding rate
+    pub cr: CodingRate,
+}
+
+/// Spreading factor for LoRa mode
+#[derive(Copy, Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[non_exhaustive]
```

Comment:
Is `#[non_exhaustive]` necessary here and above `CodingRate`?

Reply:
it means that adding additional spreading factors / coding rates in the future (what if we missed one!) would not be a breaking change and forces consumers to handle the unsupported case, i think it's worth it.